dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/fileresource/DefaultImageProcessingService.java

```diff
@@ -77,7 +78,8 @@ public Map<ImageFileDimension, File> createImages( FileResource fileResource, Fi
                     continue;
                 }
 
-                ImageSize size = IMAGE_FILE_SIZES.get( dimension );
+                ImageSize size = Optional.ofNullable( IMAGE_FILE_SIZES.get( dimension ) )
```

Comment:
I would go for a good old if/else instead of an Optional in cases like this. I find it easier to read 😅. Also wrapping something in an Optional to be unwrapped immediately does not make much sense to me if it does not improve readability.

```java
ImageSize size = IMAGE_FILE_SIZES.get( dimension );
if (size == null) {
    size = new ImageSize( 0, 0 );
}
```